### PR TITLE
Remove unused functionality from the visual metrics portable script

### PR DESCRIPTION
**Pull request by soulgalore**
Remove the following:
* --gray (remove gray frames that was an old fix for a Chrome bug)
* --timeline that synced with trace from Chrome
* The functionality to use a video with multiple navigations and split that
* Remove white frames at the end
* Remove that the navigation could start white
* Remove the possibility to trim the end of the video
*  --findstart
* --viewporttime
* --white option
* --forceblank
* Gzip option for Python 2
* --histogram option
* Removing remove frames before first orange
* --maxframes
* --notification
* functionality to generate a new video from the frames

### CODE REVIEW

**Comment by soulgalore** [comment]
@gmierz I'm gonna push more changes this weekend where I try to slimline the script. 

I have question: Do you use the script outside of Browsertime at Mozilla? I'm thinking about the ffprobe code and "knowing" if we test on mobile, when we run inside of browsertime that could be a flag (because we always know), that way we could probably sidestep using ffprobe?

**Comment by gmierz** [comment]
With the new script, that sounds good! It was more difficult for us to do that with the old script because of how we were using it.

**Comment by gmierz** [comment]
@soulgalore another thing you can get rid of is the Python2 checks. This script will only work with Python3.

**Comment by soulgalore** [comment]
@gmierz I think I'm ready for a first look. Kind of shitty review with a lot of removes. I've been verifying that the functionality works and I removed things I know is not used by Browsertime. If this is ok for you, my next move is to tune some option so that they are used by default and then see if its meaningful to split the file into multiple files to make it easier to understand and add some better tests so we can run it standalone. It would be great to have a collection of videos that we test, where we test some edge cases.

**Comment by gmierz** [comment]
Awesome! I'll review it on Monday.

Can we hold off on changing the default options and/or behaviours? I'd like to get this running in our CI before making more changes.

**Comment by soulgalore** [comment]
Works great, I'll start by adding tests then.

**Comment by gmierz** [comment]
@soulgalore I haven't forgotten about this! I'm just getting things working on our end with the existing script and fixing up some more bugs/issues while I'm at it.

**Comment by gmierz** [comment]
I've triggered a set of tests here with this patch: https://treeherder.mozilla.org/jobs?repo=try&revision=29a21fd13047493de91c05bec84d9f374339b05a

**Comment by gmierz** [comment]
@soulgalore this patch seems to have broken something for us. Here's a comparison of the metrics with and without your patch (the right side is data using this patch): https://treeherder.mozilla.org/perfherder/compare?originalProject=mozilla-central&originalRevision=2f240882d907fa71545869bf60a8a47aa69ae4d1&newProject=try&newRevision=29a21fd13047493de91c05bec84d9f374339b05a&framework=13&page=1&showOnlyComparable=1

File: browsertime/visualmetrics-portable.py [hunk 1]
```diff
@@ -414,34 +403,15 @@ def video_to_frames(
...
-                        )
-                    if multiple and orange_file is not None:
-                        directories = split_videos(directory, orange_file)
-                    else:
-                        directories = [directory]
+                    directories = [directory]
                     for dir in directories:
-                        trim_video_end(dir, trim_end)
                         if orange_file is not None:
-                            remove_frames_before_orange(dir, orange_file)
```
Comment (gmierz) [inline browsertime/visualmetrics-portable.py:428]: This method should stay or be combined with `remove_orange_frames`

**Review by gmierz — CHANGES_REQUESTED** [review]

File: browsertime/visualmetrics-portable.py (hunk 1 above)
Comment (soulgalore) [reply browsertime/visualmetrics-portable.py:428]: Ok, I can add that but how do that work on your side? Usually the orange is added first, then the recording is started?

Comment (soulgalore) [reply browsertime/visualmetrics-portable.py:428]: I've made the change but if I remember correctly the workflow is like this:
- Add the orange frame
- wait 
- start the video
 
I wonder than if that wait time is too short since it affected your tests, or is it another use case?

Comment (soulgalore) [reply browsertime/visualmetrics-portable.py:433]: But I think that do not matter now, I added it so I hope it works when you retry :) Then we can look at the logic later.

**Comment by gmierz** [comment]
New try run with the latest changes: https://treeherder.mozilla.org/jobs?repo=try&revision=dcfa4dc97e071d47ba8744e825c63308455c3563

**Comment by soulgalore** [comment]
It looks like it worked right @gmierz ?

**Comment by gmierz** [comment]
I had to retest because the last push had a lot of noisy data but the data looked really good in general. Only a couple metrics were 20-25% different but I think those are noise.

Here's a new test run for mobile: https://treeherder.mozilla.org/jobs?repo=try&revision=248e1237c260823f2ede62b9ec2b8b5f4114092c

Here's a new test run for desktop: https://treeherder.mozilla.org/#/jobs?repo=try&revision=79c8fa40c761824c7517f1f50abb4edb003c2811

I'll come back with a review by the end of my day.

**Comment by gmierz** [comment]
Desktop looks great, here's a comparison of the numbers: https://treeherder.mozilla.org/perfherder/compare?originalProject=mozilla-central&originalRevision=f3aca6a6c4a43d3485a935380dee6c0e8a3b7fda&newProject=try&newRevision=79c8fa40c761824c7517f1f50abb4edb003c2811&framework=13&page=1&showOnlyComparable=1

The bigger differences are noise. Still waiting for android to finish.

File: browsertime/visualmetrics-portable.py [hunk 2]
```diff
@@ -714,7 +634,7 @@ def find_video_viewport(
     video,
     directory,
     find_viewport,
-    viewport_time,
+
```
[the remark this answers is not in the document]
Comment (soulgalore) [reply browsertime/visualmetrics-portable.py]: Thank, fixed!

**Comment by gmierz** [comment]
Mobile tests have started completing and I don't see any major issues: https://treeherder.mozilla.org/perfherder/compare?originalProject=mozilla-central&originalRevision=f3aca6a6c4a43d3485a935380dee6c0e8a3b7fda&newProject=try&newRevision=248e1237c260823f2ede62b9ec2b8b5f4114092c&framework=13&page=12&showOnlyComparable=1

The larger differences are all noise.

**Review by gmierz — APPROVED** [review]

**Comment by gmierz** [comment]
@soulgalore r+ thanks for doing this! :)
